Toms-Storage-1206/src/shared/java/com/tom/storagemod/gui/AbstractStorageTerminalScreen.java

```diff
@@ -259,7 +257,16 @@ protected void updateSearch() {
 				for (int i = 0;i < getMenu().itemListClient.size();i++) {
 					StoredItemStack is = getMenu().itemListClient.get(i);
 					if (is != null && is.getStack() != null) {
-						String dspName = searchMod ? BuiltInRegistries.ITEM.getKey(is.getStack().getItem()).getNamespace() : is.getStack().getHoverName().getString();
+						String dspName;
+						if (searchMod) {
+							dspName = BuiltInRegistries.ITEM.getKey(is.getStack().getItem()).getNamespace();
+						} else if (searchComponent && !is.getStack().getComponentsPatch().isEmpty()) {
+							dspName = is.getStack().getComponentsPatch().toString();
```

Comment:
You have to turn components into NBT with Codecs (See `ItemStack:save`) then pretty print it into a string (NbtUtils method name changed).

Reply:
In the end, I opted for using the Json Ops for proper serialization, which imo are more reliable (as in, less likely to change with game versions)